### PR TITLE
feat(explorer): build docker image on pull request + refactor dockerfile for ipfs

### PR DESCRIPTION
**Pull request by mkjmdski**
closes: https://github.com/vegaprotocol/devops-infra/issues/1761

### CODE REVIEW

File: .github/workflows/publish-docker-containers.yml [hunk 1]
```diff
@@ -114,13 +119,16 @@ jobs:
...
+          # push: ${{ inputs.publish || startsWith(github.ref, 'refs/tags/') }}
+          file: dockerfiles/Dockerfile.cra.dist
+          build-args: |
+            APP=${{ matrix.app }}
+            NODE_VERSION=${{ steps.tags.outputs.npmVersion }}
+          # platforms: ${{ inputs.archs || 'linux/amd64, linux/arm64' }}
           tags: |
-            vegaprotocol/${{ matrix.app }}:latest
-            vegaprotocol/${{ matrix.app }}:${{ steps.tags.outputs.version }}
+            ghcr.io/vegaprotocol/frontend/${{ matrix.app }}:latest
```
Comment (daniel1302) [inline .github/workflows/publish-docker-containers.yml:130]: Why we are not publishing to dockerhub anymore?

File: .github/workflows/publish-docker-containers.yml [hunk 2]
```diff
@@ -23,25 +24,25 @@ name: Publish docker containers
         required: false
         type: choice
         options:
-          - '["explorer", "token", "trading"]'
+          # - '["explorer", "token", "trading"]'
           - '["explorer"]'
```
Comment (daniel1302) [inline .github/workflows/publish-docker-containers.yml:28]: Are we stopping builds for token and trading app?

File: .github/workflows/publish-docker-containers.yml [hunk 3]
```diff
@@ -61,7 +62,7 @@ jobs:
         uses: docker/setup-buildx-action@v2
 
       - name: Login to DockerHub
-        if: ${{ inputs.publish || startsWith(github.ref, 'refs/tags/') }}
+        # if: ${{ inputs.publish || startsWith(github.ref, 'refs/tags/') }}
```
Comment (daniel1302) [inline .github/workflows/publish-docker-containers.yml:65]: Why we need login for dockerhub when we are publishing to github repo?

**Review by daniel1302 — COMMENTED** [review]
Fix the PR name to pass the following build + docker sanity is flailing

image

File: .github/workflows/publish-docker-containers.yml (hunk 1 above)
Comment (mkjmdski) [reply .github/workflows/publish-docker-containers.yml:121]: we will be using this just internally at the moment -> so ghcr

File: .github/workflows/publish-docker-containers.yml (hunk 2 above)
Comment (mkjmdski) [reply .github/workflows/publish-docker-containers.yml:28]: yeah, we will later switch to `nx` (https://github.com/vegaprotocol/frontend-monorepo/blob/develop/tools/ipfs-deploy.js#L106-L108) to determine which apps have been modified, this is for now just a PoC

File: .github/workflows/publish-docker-containers.yml (hunk 3 above)
Comment (mkjmdski) [reply .github/workflows/publish-docker-containers.yml]: I changed to login to ghcr.io instead

**Pull request merged**